test/zod-collection.test.ts

```diff
@@ -25,6 +26,23 @@ test('Documents should be parsed correctly on insert', async () => {
     zodCol.insertOne({ a: 0, b: 'string', numbers: [0] })
   ).resolves.toBeTruthy()
 
+  const id = new ObjectId()
+  const { insertedId } = await zodCol.insertOne({
+    a: 0,
+    b: 'string',
+    numbers: [0],
+    _id: id,
```

Comment:
We should also assert that a wrong input fails here both type wise and at runtime. 
We can add a test where we pass a string to the `_id` field which can be a common mistake